pygeoapi/provider/xarray_.py

```diff
@@ -476,6 +492,71 @@ def get_time_coverage_duration(self):
 
         return ', '.join(times)
 
+    def _parse_grid_mapping(self):
+        """
+        Identifies grid_mapping.
+
+        :returns: name of xarray data variable that contains CRS information.
+        """
+        LOGGER.debug('Parsing grid mapping...')
+        spatiotemporal_dims = (self.time_field, self.y_field, self.x_field)
+        LOGGER.debug(spatiotemporal_dims)
+        grid_mapping_name = None
+        for var_name, var in self._data.variables.items():
+            if all(dim in var.dims for dim in spatiotemporal_dims):
+                try:
+                    grid_mapping_name = self._data[var_name].attrs['grid_mapping']  # noqa 
+                    LOGGER.debug(f'Grid mapping: {grid_mapping_name}')
+                except KeyError as err:
+                    LOGGER.debug(err)
+                    LOGGER.debug('No grid mapping information found.')
+        return grid_mapping_name
+
+    def _parse_storage_crs(
+        self,
+        provider_def: dict
+    ) -> pyproj.CRS:
+        """
+        Parse the storage CRS from an xarray dataset.
+
+        :param provider_def: provider definition
+
+        :returns: `pyproj.CRS` instance parsed from dataset
+        """
+        storage_crs = None
+
+        try:
+            storage_crs = provider_def['storage_crs']
+            crs_function = pyproj.CRS.from_user_input
+        except KeyError as err:
+            LOGGER.debug(err)
+            LOGGER.debug('No storage_crs found. Attempting to parse the CRS.')
+
+        if storage_crs is None:
+            grid_mapping = self._parse_grid_mapping()
+            if grid_mapping is not None:
+                storage_crs = self._data[grid_mapping].attrs
+                crs_function = pyproj.CRS.from_cf
+            elif 'crs' in self._data.variables.keys():
+                storage_crs = self._data['crs'].attrs
+                crs_function = pyproj.CRS.from_dict
+            else:
+                storage_crs = DEFAULT_STORAGE_CRS
+                crs_function = get_crs_from_uri
+                LOGGER.debug('Failed to parse dataset CRS. Assuming WGS84.')
+
+        LOGGER.debug(f'Parsing CRS {storage_crs} with {crs_function}')
+        try:
+            crs = crs_function(storage_crs)
+        except CRSError as e:
```

Comment:
`except CRSError as err:` (we typically use `err`.  Sorry for the consistency nitpick).